### PR TITLE
Initialization: get explicit annotations from Tree

### PR DESCRIPTION
**Pull request by smillst**
AnnotatedTypeFactory#fromMember should not  be used to get explicit annotations.

### CODE REVIEW

File: checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -633,6 +635,24 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
...
 
+    /**
+     * Returns the annotations explicitly written on a constructor result.
+     * Callers should check that {@code constructorDeclaration} is in fact a declaration
+     * of a constructor.
+     *
+     * @param constructorDeclaration Declaration tree of constructor
+     * @return set of annotations explicit on the resulting type of the constructor
+     */
+    public Set<AnnotationMirror> getExplicitAnnotationsOnConstructorResult(MethodTree constructorDeclaration) {
```
Comment (wmdietl) [inline checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java]: Should this method be moved to `AnnotationUtils`? Do we have `getExplicitAnnotationsOnXXX` for other XXX?
It looks like the method could be static and doesn't seem initialization specific.

Comment (smillst) [inline checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java]: Good point.  I moved the method to AnnotationUtils.  InternalUtils has these methods which are similar:
````
List<? extends AnnotationMirror> annotationsFromTree(AnnotatedTypeTree node)
List<? extends AnnotationMirror> annotationsFromTree(TypeParameterTree node)
List<? extends AnnotationMirror> annotationsFromArrayCreation(NewArrayTree node, int level)  
````

File: javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java [hunk 2]
```diff
@@ -556,4 +559,22 @@ public static Name getElementValueClassName(AnnotationMirror anno, CharSequence
...
+
+    /**
+     * Returns the annotations explicitly written on a constructor result.
+     * Callers should check that {@code constructorDeclaration} is in fact a declaration
+     * of a constructor.
+     *
+     * @param constructorDeclaration Declaration tree of constructor
+     * @return set of annotations explicit on the resulting type of the constructor
+     */
+    public static Set<AnnotationMirror> getExplicitAnnotationsOnConstructorResult(MethodTree constructorDeclaration) {
```
Comment (wmdietl) [inline javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java]: Would it make sense to move this to https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/ConstructorReturnUtil.java ?
That class only has a single method and seems rather underused.

Comment (smillst) [inline javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java]: I would prefer to leave it in AnnotationUtils where it is more likely to be discovered and used.  I also think when `@TargetLocations` are enforced (Issue #515), we will want to add methods to get explicit annotations from other trees.

Comment (wmdietl) [inline javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java]: Can we just move the single method from ConstructorReturnUtil to AnnotationUtils? It seems counter-intuitive to only have one of the methods about constructor returns in a separate class.

**Comment by wmdietl** [comment]
Looks good to merge.

**Pull request merged**